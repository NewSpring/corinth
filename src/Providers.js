import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import ExternalLinkProvider from './linking/Provider';
import track from './analytics';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <ExternalLinkProvider navigate={NavigationService.navigate}>
          <MediaPlayerProvider>
            <AnalyticsProvider trackFunctions={[track]}>
              <LiveProvider>
                <Providers
                  themeInput={customTheme}
                  iconInput={customIcons}
                  {...props}
                />
              </LiveProvider>
            </AnalyticsProvider>
          </MediaPlayerProvider>
        </ExternalLinkProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
