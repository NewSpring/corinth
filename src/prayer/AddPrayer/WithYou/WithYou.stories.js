import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';

import WithYou from '.';

storiesOf('prayer/AddPrayer/WithYou', module).add('default', () => <WithYou />);
