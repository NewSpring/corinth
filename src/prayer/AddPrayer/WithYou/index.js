import React, { memo } from 'react';
import { SafeAreaView } from 'react-navigation';
import {
  ModalView,
  FlexedView,
  PaddedView,
  styled,
  Button,
  ButtonLink,
  BodyText,
  Icon,
  H1,
  withTheme,
  Paragraph,
} from '@apollosproject/ui-kit';

const FlexedSafeAreaView = styled(({ theme }) => ({
  flex: 1,
  paddingHorizontal: theme.sizing.baseUnit,
}))(SafeAreaView);

const MessageView = styled({
  flex: 4,
  justifyContent: 'center',
})(FlexedView);

const BrandIcon = withTheme(({ theme }) => ({
  name: 'brand-icon',
  size: theme.sizing.baseUnit * 4,
  fill: theme.colors.primary,
}))(Icon);

const Title = styled(({ theme }) => ({
  marginVertical: theme.sizing.baseUnit,
  color: theme.colors.primary,
}))(H1);

const Footer = styled({
  justifyContent: 'flex-end',
})(FlexedView);

const FooterAltOption = styled({
  alignSelf: 'center',
})(PaddedView);

const FooterText = styled(({ theme }) => ({
  color: theme.colors.tertiary,
}))(BodyText);

const WithYou = memo(({ ...props }) => (
  <ModalView {...props} onClose={() => props.navigation.popToTop()}>
    <FlexedSafeAreaView>
      <MessageView>
        <BrandIcon />
        <Title>Thank you for sharing</Title>
        <Paragraph>
          <BodyText>
            “This is the confidence we have in approaching God: that if we ask
            anything according to his will, he hears us” (1 John 5:14).
          </BodyText>
        </Paragraph>
        <Paragraph>
          <BodyText>We’re praying for you and believing with you.</BodyText>
        </Paragraph>
      </MessageView>
      <Footer>
        <Button
          title="Pray for others"
          onPress={() => props.navigation.navigate('PrayerList')}
        />
        <FooterAltOption>
          <ButtonLink onPress={() => props.navigation.pop()}>
            <FooterText>Add another prayer request</FooterText>
          </ButtonLink>
        </FooterAltOption>
      </Footer>
    </FlexedSafeAreaView>
  </ModalView>
));

WithYou.displayName = 'WithYou';

WithYou.navigationOptions = {
  header: null,
};

export default WithYou;
