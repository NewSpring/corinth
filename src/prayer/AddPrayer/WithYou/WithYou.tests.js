import React from 'react';
import renderer from 'react-test-renderer';
import Providers from 'newspringchurchapp/src/Providers';

import WithYou from '.';

describe('The WithYou component', () => {
  it('should render', () => {
    const navigation = { pop: jest.fn(), navigate: jest.fn() };
    const tree = renderer.create(
      <Providers>
        <WithYou navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
