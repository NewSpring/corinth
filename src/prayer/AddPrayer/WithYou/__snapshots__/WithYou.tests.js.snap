// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The WithYou component should render 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294967295,
      4294506484,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "borderRadius": 0,
      "flex": null,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "justifyContent": "center",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={64}
        bbWidth={64}
        height={64}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            null,
            Object {
              "flex": 0,
              "height": 64,
              "width": 64,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={64}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M12 22C5.2 22 2 17.52 2 12S6.48 2 12 2s10 3.33 10 10v10H12zM7.84 8v9.08h3.12V12.3c0-1.14.6-1.8 1.63-1.8.53 0 1.03.27 1.23.68.17.33.2.5.2 1.4v4.5h3.1v-5.65c0-1.2-.2-1.82-.8-2.45C15.7 8.33 14.86 8 13.8 8H7.84z"
            fill={
              Array [
                0,
                4285246531,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#6bac43",
            "fontFamily": "Colfax-Black",
            "fontSize": 43,
            "lineHeight": 49.45,
            "marginVertical": 16,
          }
        }
      >
        Thank you for sharing
      </Text>
      <View
        style={
          Object {
            "paddingBottom": 17.28,
          }
        }
      >
        <Text
          bold={false}
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Regular",
              "fontSize": 16,
              "lineHeight": 26,
            }
          }
        >
          “This is the confidence we have in approaching God: that if we ask anything according to his will, he hears us” (1 John 5:14).
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingBottom": 17.28,
          }
        }
      >
        <Text
          bold={false}
          italic={false}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Regular",
              "fontSize": 16,
              "lineHeight": 26,
            }
          }
        >
          We’re praying for you and believing with you.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          bordered={false}
          disabled={false}
          pill={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#6bac43",
              "borderColor": "#6bac43",
              "borderRadius": 48,
              "borderWidth": 2,
              "elevation": 2,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "opacity": 1,
              "overflow": "hidden",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "Colfax-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            Pray for others
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#6bac43",
            }
          }
        >
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#2a4930",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
              }
            }
          >
            Add another prayer request
          </Text>
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 16,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={32}
      bbWidth={32}
      fill="#303030"
      height={32}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 32,
            "width": 32,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={32}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4281348144,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        borderRadius={24}
        iconPadding={8}
        style={
          Object {
            "backgroundColor": Object {
              "color": Array [
                48,
                48,
                48,
              ],
              "model": "rgb",
              "valpha": 0.7,
            },
            "padding": 8,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={16}
          bbWidth={16}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={16}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M13.07 12l7.2-7.2.5-.5c.3-.3.3-.78 0-1.08-.3-.3-.76-.3-1.06 0l-.5.5-7.2 7.2-7.2-7.2-.5-.5c-.3-.3-.77-.3-1.07 0-.3.3-.3.77 0 1.07l.5.5 7.2 7.2-7.7 7.7c-.3.3-.3.77 0 1.07.3.3.77.3 1.07 0l7.7-7.7 7.7 7.7c.3.3.78.3 1.08 0 .3-.3.3-.77 0-1.07l-7.7-7.7z"
              fill={
                Array [
                  0,
                  4294967295,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;
